File: src/main/java/org/dedda/bratwurst/lang/BWString.java
package org.dedda.bratwurst.lang;

/**
 * Created by dedda on 10/15/15.
 *
 * @author dedda
 */
public class BWString extends BWObject {

    private String value;

    public BWString(String value) {
        super(new BWStringClass());
        this.value = value;
    }

    public String getStringValue() {
        return value;
    }

    @Override
    public String getValueType() {
        return "string";
    }

    @Override
    public int getIntValue() {
        if (value.matches("\\-?\\d+")) {
            return Integer.parseInt(value);
        }
        return 0;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        BWString bwString = (BWString) o;

        return !(value != null ? !value.equals(bwString.value) : bwString.value != null);

    }

}

File: src/main/java/org/dedda/bratwurst/lang/BWString.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.ValueType.STRING

/**
 * Created by dedda on 10/15/15.
 *
 * @author dedda
 */
class BWString(val stringValue: String) : BWObject(BWClass.getClassForName("string")) {

    override fun getValueType() = STRING

    override fun getIntValue(): Int {
        return if (stringValue.matches(Regex("-?\\d+"))) {
            stringValue.toInt()
        } else 0
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val bwString = other as BWString
        return stringValue == bwString.stringValue
    }

    override fun hashCode(): Int {
        return stringValue.hashCode()
    }

}